web-wallet/src/lib/stores/walletStore.js

```diff
@@ -51,15 +51,30 @@ const { set, subscribe, update } = walletStore;
 const treasury = new WalletTreasury();
 const bookkeeper = new Bookkeeper(treasury);
 
-/** @type {<T>(action: (...args: any[]) => Promise<T>) => Promise<T>} */
-const effectfulAction = (action) => sync().then(action).finally(updateBalance);
-
 const getCurrentAccount = () => get(walletStore).currentProfile?.account;
 const getCurrentAddress = () => get(walletStore).currentProfile?.address;
 
 /** @type {(...args: any) => Promise<void>} */
 const asyncNoopFailure = () => Promise.reject(new Error("Not implemented"));
 
+/** @type {(txInfo: TransactionInfo) => void} */
+const observeTxRemoval = (txInfo) => {
+  networkStore.connect().then((network) =>
+    network.transactions
+      .withId(txInfo.hash)
+      .once.removed()
+      .then(() => sync())
+      .finally(updateBalance)
+  );
```

Comment:
Very clean! ✨

Reply:
Thanks, although it's not because of me in this case!